### ISSUE TITLE
Ability to update "pending" DNS zones

### ISSUE THREAD

**Issue by wzball** [favonia/cloudflare-ddns#194]
Hi, thanks for your work.

I have a domain in another registrar and the NS's pointing to Cloudflare. I'd set these up as websites rather than as a managed domain.

I'm expecting that that's the reason why I get the following messages.

```
Failed to find the zone of "domain.com"
Failed to (fully) update A records of "domain.com"
```

Is there a method to work with Cloudflare Websites instead of Domains ?

Thanks!

**Comment by favonia** [comment]
Hi, thanks for using the updater. You can definitely use this updater with a domain registrar other than Cloudflare, and only using Cloudflare as the name server. However, even if the Cloudflare is used only as a name server, it should still have a DNS _zone_, which represents the portion that the name server is in charge of. The problem here is that the updater failed to automatically detect the correct zone. There are many possible reasons, from a simple typo to token miscommunication. I need to know more about your setup. Could you possibly run this on your machine to list all zones under your account?
```sh
curl -X GET "https://api.cloudflare.com/client/v4/zones" \
    -H "Authorization: Bearer YOUR-CLOUDFLARE-API-TOKEN" \
    -H "Content-Type: application/json"
```
The output might contain sensitive information so please do not post it here, but it will be great if you can double-check that there is a zone whose name is the suffix of the domain you are updating. For example, if you want to update the IP of the domain "www.abc.example.org", you should see a zone "www.abc.example.org" or "abc.example.org" or "example.org" (or theoretically "org" or even the root, but I assume you are from the organizations managing those zones :laughing:). If the output is too difficult to read, you can use tools like `jq` to reformat it.

**Comment by favonia** [comment]
@wzball Hi, did you get the chance to work on this? I am very interested in finding and fixing the bug (if any).

**Comment by wzball** [comment]
Sorry, just getting to it now.

**Comment by wzball** [comment]
I've found an issue my side, the site name server hasn't updated in Cloudflare, even though the other one has. Not sure at this time why the NS services hasn't been picked up by Cloudflare as they are being referred to when I do an external NS domain lookup. I'll wait till this has completed and then check-in here. Thanks.

![image](https://user-images.githubusercontent.com/53862711/184458915-77525528-09be-4f1c-9e54-945edf617987.png)

![image](https://user-images.githubusercontent.com/53862711/184459122-fc93aa22-cae9-4def-b8f9-aaab1aead59c.png)

**Comment by wzball** [comment]
Here's the output of the curl command.

![image](https://user-images.githubusercontent.com/53862711/184459564-4ba05601-bd25-48e4-bac3-81bd08d49fd8.png)

**Comment by favonia** [comment]
I see. The reason is that the updater checks whether a zone is "active". Your zone is still "pending", and thus the updater skipped the zone. I believe you still need to resolve the "Pending Nameserver Update", but I could relax some of the safety checks in the updater so that you could update the DNS records before your issue with Cloudflare is resolved.

I recommend checking out the forum https://community.cloudflare.com as there are many friendly and helpful people there for your "Pending Nameserver Update" issue. By the way, I also recommend covering the sensitive information with solid shapes instead of blurring it, because it is actually possible to undo the blurring. :slightly_smiling_face:

tl;dr: I will update the tool so that you can run the updater, but you still have work to do to really solve the problem.

**Comment by favonia** [comment]
@wzball The code in the `main` branch should work. Could you possible try it out (or using the Docker image `favonia/cloudflare-ddns:edge`)? You would see a warning about the zone not being active, but otherwise it should work. I prefer having your confirmation before releasing a new version.

**Comment by wzball** [comment]
Thanks for your responses @favonia ... turns out the registrar that I'm using has a separate section to modify NS records, even though it has a web UI to add NS records (same area for A/CNAME records) to the Zone. :/ not sure what to make of that. It wasn't very clear. either way it's working fine for me. Layer 8 issue - not setting the NS in the correct area.

**Comment by favonia** [comment]
Glad to hear that it worked. Out of curiosity, was your problem solved by fixing the root issue (fully activating your zone) or using the newer version (updating the zone even if it is not fully activated)? By the way, I believe the UI to add NS records is for you to add authoritative nameservers for subdomains.
